util/ci-run-file-creator.py

```diff
@@ -286,6 +333,15 @@ def update_host():
                     endpoint["host"] = args.host
                 else:
                     log.debug("No host to update for %s endpoint %d" % (args.endpoint, endpoint_idx))
+        case "remotehosts":
+            endpoint = run_file["endpoints"][0]
```

Comment:
why endpoints index 0 ? Isn't this an array of endpoints ?

Reply:
So a little bit of a cheat/assumption here.  Yes, the JSON is capable of hosting many endpoints in the endpoints array -- BUT, in CI only a single endpoint type is being processed at any one time AND the remotehosts endpoint does not support multiple invocations of itself in the same JSON.  So, by process of elimination the remotehosts endpoint must be at array index 0.

The reason the remotehosts endpoint does not support multiple invocations of itself comes down to 2 reasons:

1. It should not ever need multiple invocations.  Since it supports multiple remotes and each remote can have it's own set of endpoint settings there is no need to have multiple endpoints of the same remotehosts type.  If at some point we find a reason to need this, I view that as bug that needs to be fixed.
2. This is probably the more important of the two reasons.  The remotehosts endpoint dedups the list of remotes into a unique set that it uses to perform certain operations that should only be performed once per remote such as pulling container images, performing container image cache management, and determining where profiler engines should be run.  If multiple remotehosts endpoints where specified it would not be able to properly dedup the remotes across the multiple invocations and would then perform these operations multiple times on the same remotes.  This was one of the scalability problems detected with the current remotehost endpoint -- certain operations such as pulling container images would breakdown when being done too many times in parallel on the same remote.